presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -282,6 +262,61 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, Co
     }
 
     /**
+     * Verify that the partition schema is backward compatible with the table schema. Either
+     * adding or dropping columns from the end of the table without modifying existing partitions is
+     * allowed, but every column that exists in both the table and partition must have the same type.
+     *
+     * If the type is a nested type (i.e. a struct), verify all partition struct fields exist in the
+     * table struct fields in with the same name, type and order. Af that condition is met, additional
+     * table struct fields are permitted.
+     *
+     * @throws PrestoException if the schema has not evolved in a supported way
+     */
+    private void verifySchemaEvolution(SchemaTableName tableName, List<FieldSchema> tableColumns,
```

Comment:
Do we want to move forward with this PR or PR #4714 (or both)? If we want to merge this one, which could be an iteration towards a the larger #4714, then I can add integration tests to verify the other formats.